crates/db_views/notification/src/impls.rs

```diff
@@ -0,0 +1,315 @@
+  pub async fn list(
+    self,
+    pool: &mut DbPool<'_>,
+    my_user_view: &LocalUserView,
+  ) -> LemmyResult<Vec<NotificationView>> {
+    let conn = &mut get_conn(pool).await?;
+
+    let item_creator = person::id;
+    let recipient = notification::recipient_id;
+
+    let mut query =
+      NotificationView::joins(my_user_view.person.id, my_user_view.person.instance_id)
+        .select(NotificationViewInternal::as_select())
+        .into_boxed();
+
+    if !self.no_limit.unwrap_or_default() {
+      let limit = limit_fetch(self.limit)?;
+      query = query.limit(limit);
+    }
+
+    // Filters
+    if self.unread_only.unwrap_or_default() {
+      query = query
+        // The recipient filter (IE only show replies to you)
+        .filter(notification::recipient_id.eq(my_user_view.local_user.id))
+        .filter(
+          notification::read
+            .eq(false)
+            // If its unread, I only want the messages to me
+            .or(private_message::read.eq(false)),
+        );
+    } else {
+      // A special case for private messages: show messages FROM you also.
+      // Use a not-null checks to catch the others
+      query = query.filter(
+        recipient.eq(my_user_view.local_user.id).or(
+          notification::private_message_id.is_not_null().and(
+            recipient
+              .eq(my_user_view.local_user.id)
+              .or(item_creator.eq(my_user_view.person.id)),
+          ),
+        ),
+      );
+    }
+
+    if !(self.show_bot_accounts.unwrap_or_default()) {
+      query = query.filter(not(person::bot_account));
+    };
+
+    // Dont show replies from blocked users or instances
+    query = query
+      .filter(person_actions::blocked_at.is_null())
+      .filter(instance_actions::blocked_at.is_null());
+
+    if let Some(type_) = self.type_ {
+      query = match type_ {
+        InboxDataType::All => query,
+        InboxDataType::Reply => query.filter(notification::kind.eq(NotificationTypes::Reply)),
+        InboxDataType::Mention => query.filter(notification::kind.eq(NotificationTypes::Mention)),
+        InboxDataType::PrivateMessage => {
+          query.filter(notification::kind.eq(NotificationTypes::PrivateMessage))
+        }
+      }
+    }
+
+    // Sorting by published
+    let paginated_query = paginate(
+      query,
+      SortDirection::Desc,
+      self.cursor_data,
+      None,
+      self.page_back,
+    )
+    .then_order_by(notification_keys::published_at)
+    // Tie breaker
+    .then_order_by(notification_keys::id);
+
+    let res = paginated_query
+      .load::<NotificationViewInternal>(conn)
+      .await?;
+
+    Ok(res.into_iter().filter_map(map_to_enum).collect())
+  }
+}
+
+fn map_to_enum(v: NotificationViewInternal) -> Option<NotificationView> {
+  let data = if let (Some(comment), Some(post), Some(community)) =
+    (v.comment, v.post.clone(), v.community.clone())
+  {
+    NotificationData::Comment {
+      comment,
+      post,
+      community,
```

Comment:
Check the previous version, but this needs to be the full views. Or you can look at the other combined views.